### PR TITLE
feat: make zip support larger files

### PR DESCRIPTION
**Pull request by leangseu-edx**
Allow zip to support larger files. As far as I can tell, this can support is based on the free space in the computer. 50% for chrome and 20% for FireFox. 

I have tested individual small files that combined update to 6GB and for the largest individual file it can compress without error is approximately 2GB. 

### Test Instruction
- Download files you want to download test with
- Place those file in `public/samples/{file}`
- Replace [this line](https://github.com/edx/frontend-app-ora-grading/blob/e847f5f1d6c40e4838d4162f7158fa37d6ac0878/src/data/redux/thunkActions/download.js#L70) with
```js
  const files = (new Array(20).fill(0)).map((_, i) => ({
    downloadUrl: `samples/sample-${i%6}.mp4`,
    name: `sample-${i}.mp4`,
    description: 'some description',
    size: 5,
  }));
```
- Rename those files to `sample-{number}.mp4`
- Modify the script accordingly
- Go to one of the ora that has `Download files`
- Click download

### CODE REVIEW

File: src/setupTest.js [hunk 1]
```diff
@@ -99,3 +99,5 @@ jest.mock('@edx/paragon/icons', () => ({
 jest.mock('data/constants/app', () => ({
   locationId: 'fake-location-id',
 }));
+
+jest.mock('@zip.js/zip.js', () => ({}));
```
Comment (leangseu-edx) [inline src/setupTest.js:110]: `jest` isn't happy with the default `@zip.js/zip.js`. This empty mock is to help `jest` ignoring the module import.

File: src/data/redux/thunkActions/download.js [hunk 2]
```diff
@@ -31,13 +31,15 @@ export const zipFileName = () => {
...
-  const zip = new JSZip();
-  zip.file('manifest.txt', module.genManifest(files));
-  blobs.forEach((blob, i) => zip.file(files[i].name, blob));
-  return zip.generateAsync({ type: 'blob' }).then(
-    zipFile => FileSaver.saveAs(zipFile, module.zipFileName()),
-  );
+export const zipFiles = async (files, blobs) => {
+  const zipWriter = new zip.ZipWriter(new zip.BlobWriter('application/zip'));
+  await zipWriter.add('manifest.txt', new zip.TextReader(module.genManifest(files)));
+  await Promise.all(blobs.map(async (blob, i) => zipWriter.add(files[i].name, new zip.BlobReader(blob), {
```
Comment (muselesscreator) [inline src/data/redux/thunkActions/download.js]: Please split this into multiple lines for readability.

File: src/setupTest.js (hunk 1 above)
Comment (muselesscreator) [reply src/setupTest.js:110]: ++

File: src/data/redux/thunkActions/download.test.js [hunk 3]
```diff
@@ -44,29 +57,29 @@ describe('download thunkActions', () => {
   const getState = () => testState;
   describe('genManifest', () => {
     test('returns a list of strings with filename and description for each file', () => {
-      expect(download.genManifest(response.files)).toEqual([
-        `Filename: ${files[0].name}\nDescription: ${files[0].description}\nSize: ${files[0].size}`,
-        `Filename: ${files[1].name}\nDescription: ${files[1].description}\nSize: ${files[0].size}`,
-      ].join('\n\n'));
+      expect(download.genManifest(response.files)).toEqual(
+        [
+          `Filename: ${files[0].name}\nDescription: ${files[0].description}\nSize: ${files[0].size}`,
```
Comment (muselesscreator) [inline src/data/redux/thunkActions/download.test.js:62]: I wish we were testing this from a variable, but that is not a blocker here.

**Review by muselesscreator — COMMENTED** [review]
Minor formatting concern, but other than that looks good to me, as long as is tested.

**Review by muselesscreator — APPROVED** [review]

**Pull request merged**